modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/task/QueryTasksQueue.java

```diff
@@ -160,8 +182,9 @@ public void unblockQuery(QueryKey qryKey) {
 
             assert removed;
 
-            if (cnt.get() > 0)
-                notEmpty.signal();
```

Comment:
We removed almost all signals. With the condition `if (tasksCnt > freeThreadsCnt` in the task adding, it seems we might keep rest of the tasks unprocessed until a new one comes. If it won't, some tasks wil lstay in the queue unprocessed. WDYT?